lib/jxl/dec_frame.cc

```diff
@@ -651,6 +654,72 @@ Status FrameDecoder::ProcessACGroup(size_t ac_group_id,
   return true;
 }
 
+void FrameDecoder::PreparePipeline() {
+  if (frame_header_.nonserialized_metadata->m.num_extra_channels != 0) {
+    JXL_ABORT("Not implemented: extra channels");
+  }
+  if (frame_header_.upsampling != 1) {
+    JXL_ABORT("Not implemented: upsampling");
+  }
+  if (!frame_header_.chroma_subsampling.Is444()) {
+    JXL_ABORT("Not implemented: chroma subsampling");
+  }
+
+  RenderPipeline::Builder builder(
+      std::vector<std::pair<size_t, size_t>>{{0, 0}, {0, 0}, {0, 0}});
+
+  if (use_slow_rendering_pipeline_) {
+    builder.UseSimpleImplementation();
+  } else {
+    JXL_ABORT("Not implemented: fast pipeline");
+  }
+
+  if (frame_header_.loop_filter.gab) {
+    JXL_ABORT("Not implemented: Gaborish");
+  }
+  if (frame_header_.loop_filter.epf_iters != 0) {
+    JXL_ABORT("Not implemented: EPF");
+  }
+  if ((frame_header_.flags & FrameHeader::kPatches) != 0) {
+    JXL_ABORT("Not implemented: patches");
+  }
+  if ((frame_header_.flags & FrameHeader::kSplines) != 0) {
+    JXL_ABORT("Not implemented: splines");
+  }
+  if ((frame_header_.flags & FrameHeader::kNoise) != 0) {
+    JXL_ABORT("Not implemented: noise");
+  }
+  if (frame_header_.dc_level != 0) {
+    JXL_ABORT("Not implemented: save as dc frames");
+  }
+  if (!coalescing_) {
+    JXL_ABORT("Not implemented: skip coalescing");
+  }
+  if (dec_state_->shared->frame_header.CanBeReferenced()) {
+    JXL_ABORT("Not implemented: save as reference");
+  }
+
+  if (frame_header_.color_transform == ColorTransform::kYCbCr) {
+    JXL_ABORT("Not implemented: YCbCr");
+  } else if (frame_header_.color_transform == ColorTransform::kXYB) {
+    builder.AddStage(GetXYBStage(dec_state_->output_encoding_info));
+  }  // Nothing to do for kNone.
```

Comment:
Obviously not related to this PR, but we might actually want to do something for kNone: if the original colorspace is an enum one, there's no reason why we have to return pixels in that colorspace: we can also convert to the preferred colorspace. If we do that (also for the kYCbCr case), we can make it more likely that you'll get pixels in the preferred colorspace, which will make it more useful for applications to actually set the preferred colorspace.